include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,200 @@
+                      common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus);
+};
+
+template <class T>
+void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
+    Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+    common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus) {
+  using hpx::unwrapping;
+
+  auto executor_hp = dlaf::getHpExecutor<Backend::MC>();
+  auto executor_np = dlaf::getNpExecutor<Backend::MC>();
+
+  const SizeType m = mat_c.nrTiles().rows();
+  const SizeType n = mat_c.nrTiles().cols();
+  const SizeType mb = mat_v.blockSize().rows();
+
+  // Matrix T
+  if (m <= 1 || n == 0)
+    return;
+
+  const SizeType nr_reflector = mat_v.size().rows() - mb;
+
+  dlaf::matrix::Distribution dist_t({mb, nr_reflector}, {mb, mb});
+
+  constexpr std::size_t n_workspaces = 2;
+  common::RoundRobin<matrix::Panel<Coord::Col, T, Device::CPU>> panelsVV(n_workspaces,
+                                                                         mat_v.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Col, T, Device::CPU>> panelsW(n_workspaces,
+                                                                        mat_v.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Row, T, Device::CPU>> panelsW2(n_workspaces,
+                                                                         mat_c.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Row, T, Device::CPU>> panelsT(n_workspaces, dist_t);
+
+  // Specific for V matrix layout where last column of tiles is empty
+  const SizeType num_panel_refls = mat_v.nrTiles().cols() - 1;
+
+  for (SizeType k = num_panel_refls - 1; k >= 0; --k) {
+    bool is_last = (k == num_panel_refls - 1) ? true : false;
+
+    auto& panelVV = panelsVV.nextResource();
+    auto& panelW = panelsW.nextResource();
+    auto& panelW2 = panelsW2.nextResource();
+    auto& panelT = panelsT.nextResource();
+
+    for (SizeType i = k + 1; i < mat_v.nrTiles().rows(); ++i) {
+      // Copy V panel into VV
+      auto ik = LocalTileIndex{i, k};
+
+      // Setting VV
+      if (i == k + 1) {
+        copySingleTile(mat_v.read(ik), panelVV(ik));
+        hpx::dataflow(hpx::launch::sync, unwrapping(tile::laset<T>), lapack::MatrixType::Upper, 0.f, 1.f,
+                      std::move(panelVV(ik)));
+      }
+      else {
+        hpx::shared_future<matrix::Tile<const T, Device::CPU>> tile_v = mat_v.read(ik);
+
+        if (is_last) {
+          tile_v = splitTile(tile_v, {{0, 0}, dist_t.tileSize(GlobalTileIndex(k, k))});
+        }
+
+        panelVV.setTile(ik, mat_v.read(ik));
+      }
+
+      // Copy VV into W
+      copySingleTile(panelVV.read(ik), panelW(ik));
+    }
+
+    const GlobalTileIndex v_start{k + 1, k};
+    auto taus_panel = taus[k];
+    const SizeType taupan =
+        (is_last) ? mat_v.tileSize(GlobalTileIndex(0, m - 1)).cols() : mat_v.blockSize().cols();

```

Comment:
This size is used in multiple place (e.g. width of V and W panels). Find a more explicative name and define it at the beginning of the loop.